### PR TITLE
dotCMS/core#16794 Edit field variables is creating a new one

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.ts [hunk 1]
```diff
@@ -61,12 +62,18 @@ export class DotContentTypeFieldsVariablesTableRowComponent implements OnInit {
 
     /**
      * Focus on Key input
-     * @param {boolean} [forced]
+     * @param {Event} [$event]
      * @memberof DotContentTypeFieldsVariablesTableRowComponent
      */
-    focusKeyInput(forced?: boolean): void {
-        if (forced || this.isFieldDisabled()) {
+    focusKeyInput($event?: Event): void {
```
Comment (freddyucv) [inline src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.ts]: maybe we should change this method name

File: src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.ts [hunk 2]
```diff
@@ -115,6 +122,10 @@ export class DotContentTypeFieldsVariablesTableRowComponent implements OnInit {
         this.save.emit(this.variableIndex);
     }
 
+    private isFieldVariableUpdate($event: Event): boolean {
+        return typeof $event !== 'undefined';
```
Comment (freddyucv) [inline src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.ts]: why can it be 'undefined'?

File: src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.html [hunk 3]
```diff
@@ -1,32 +1,46 @@
...
+                </ng-template>
+                <ng-template pTemplate="output">
+                    <span *ngIf="fieldVariable.key">
+                        {{ fieldVariable.key }}
+                    </span>
+                    <span class="empty-placeholder" *ngIf="!fieldVariable.key">
+                        {{ messages['contenttypes.field.variables.key_input.placeholder'] }}
+                    </span>
+                </ng-template>
+            </p-cellEditor>
```
Comment (freddyucv) [inline src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.html]: maybe this code should be a componente, what do you think?

Comment (fmontes) [reply src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.html]: That’s a component

File: src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.ts [hunk 4]
```diff
@@ -55,19 +58,24 @@ export class DotContentTypeFieldsVariablesTableRowComponent implements OnInit {
...
      * Focus on Key input
-     * @param {boolean} [forced]
+     * @param {Event} [$event]
      * @memberof DotContentTypeFieldsVariablesTableRowComponent
      */
-    focusKeyInput(forced?: boolean): void {
-        if (forced || this.isFieldDisabled()) {
-            this.keyCell.nativeElement.click();
-        }
+    focusKeyInput($event: Event): void {
```
Comment (freddyucv) [inline src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.ts]: should chenge this method name really it does nor focus the Key Input

**Pull request merged**